apm-agent-core/src/main/java/co/elastic/apm/agent/logging/Log4jLoggerFactoryBridge.java

```diff
@@ -33,6 +35,13 @@
     private static final String FQCN = Log4jLoggerFactoryBridge.class.getName();
     private static final String PACKAGE = "co.elastic.apm.agent.sdk.logging";
 
+    public static void shutdown() {
+        LoggerContextFactory factory = LogManager.getFactory();
+        for (LoggerContext context : ((Log4jContextFactory) factory).getSelector().getLoggerContexts()) {
```

Comment:
Tests fail on this cast, please take a look

Reply:
fixed